### PR TITLE
Framework for determinisitc tests for btcstaking protocol

### PR DESCRIPTION
**Pull request by KonradStaniec**
Extends testing framework for deterministic cross module integration test to enable finality voting.

Also add nicer api to simulate operatoins mimicking how they happen in practice.

Use cases for this framework:
- cross module integration tests checking edge case scenarios
- replay tests to check determinism of particular scenarios

### CODE REVIEW

File: test/replay/btcstaking_test.go [hunk 1]
```diff
@@ -40,75 +39,307 @@ func FuzzCreatingAndActivatingDelegations(f *testing.F) {
...
-		}
+func TestNewAccountCreation(t *testing.T) {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	driverTempDir := t.TempDir()
+	replayerTempDir := t.TempDir()
+	driver := NewBabylonAppDriver(r, t, driverTempDir, replayerTempDir)
+	// first finalize at least one block
+	driver.GenerateNewBlock()
+
+	stakers := driver.CreateNStakerAccounts(5)
```
Comment (RafilxTenfen) [inline test/replay/btcstaking_test.go:70]: We should pass the random down to account creation as well, it could be in a future improvement

**Review by RafilxTenfen — APPROVED** [review]

File: test/replay/scenario_builder.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+func NewStandardScenario(driver *BabylonAppDriver) *StandardScenario {
+	return &StandardScenario{
+		driver: driver,
+	}
+}
+
+// TODO: for now scenario works for small amount of fps and stakers, as with
+// larger amound we are hitting per block gas limit, which leads to account sequence
+// nuber errors. Improve this in the future.
```
Comment (Lazar955) [inline test/replay/scenario_builder.go:21]:
```suggestion
// number errors. Improve this in the future.
```

**Review by Lazar955 — APPROVED** [review]
Nice work! Maybe add `t.Parallel()` in all unit and fuzz tests
